optimum/exporters/onnx/model_configs.py

```diff
@@ -400,7 +428,7 @@ def generate(self, input_name: str, framework: str = "pt", int_dtype: str = "int
         return int_tensor
 
 
-class BartOnnxConfig(TextSeq2SeqOnnxConfig):
+class M2M100OnnxConfig(TextSeq2SeqOnnxConfig):
```

Comment:
I'm not sure to understand the inheritance change here for M2M100/Bart/Mbart.

Reply:
we want to override `patch_model_for_export` for OnnxConfig of some architectures but not for `M2M100OnnxConfig`  see https://github.com/huggingface/optimum/blob/refactorization-decoder-ort/optimum/exporters/onnx/model_configs.py#L555-L570

Reply:
@echarlaix can you send a static link to refer to?

Reply:
discussed internally and resolved